feubot.py

```diff
@@ -4,38 +4,730 @@
+        output = "No output." if not output else output
+        await bot.say(output)
+
+
 if __name__ == "__main__":
     if "--debug" in argv:
-        bot = commands.Bot(command_prefix=['##', 'feubeta '], description='this is feubot beta.', formatter = FeubotFormatter()) 
+        bot = commands.Bot(command_prefix=['##', 'feubeta '], description='this is feubot beta.', formatter = FeubotFormatter())
     else:
-        bot = commands.Bot(command_prefix=['!', '>>', 'feubot '], description='this is feubot.', formatter = FeubotFormatter())
+        bot = commands.Bot(command_prefix=['!', '>>', 'feubot '], description="Hi I'm feubot. I'm case sensitive.", formatter = FeubotFormatter())
+
+    def trunc_to(ln, s):
+        if len(s) >= ln: return s
+        else: return s[:ln-3] + "..."
+
+    def create_embed(posts, threads, term):
+        feu_search_base = "http://feuniverse.us/search?q=%s"
+        feu_post_base = "http://feuniverse.us/t/{}/{}"
+
+        result = discord.Embed(
+                title="Search results",
+                url=feu_search_base % term,
+                description="Found %d results" % len(posts),
+                color=0xde272c)
+        for i,post in enumerate(posts[:5]):
+            result.add_field(
+                    name='Post in "%s" by %s' % (threads[i]["title"], post["name"]),
+                    value="[%s](%s)" %
+                        (trunc_to(50, post["blurb"]),
+                         feu_post_base.format(post["topic_id"], post["post_number"])),
+                    inline=False)
+        if len(posts) > 5:
+            result.set_footer(text="Truncated %d result(s)." % (len(posts)-5))
+        return result
+
+
 
     @bot.event
     async def on_ready():
         print('Logged in as')
         print(bot.user.name)
         print(bot.user.id)
         print('------')
-        await bot.change_presence(game=discord.Game(name="Reading the doc!"))
+        #await bot.change_presence(status=discord.Status.idle, activity=game)
+        await bot.change_presence(activity=discord.Game(name="Eating some beans."))
+
+    @bot.add_listener
+    async def on_command_error(error, ctx):
+        if type(error) == commands.CheckFailure:
+            pass
+        elif type(error) == commands.CommandNotFound:
+            pass
+        else:
+            print(error)#await ctx.channel.send ("ctx.message.channel, error")
 
+###################The Legend
     @bot.command()
-    async def donate():
+    async def donate(ctx):
         """you know it"""
-        await bot.say("https://donorbox.org/donate-to-circles")
+        await ctx.channel.send("https://www.patreon.com/theFEUfund")
+        await ctx.channel.send("https://donorbox.org/donate-to-circles")
 
-    token = os.environ.get('TOKEN', default=None)
-    if token is None:
-        token = open('./token').read().replace('\n','')
+##################Local
+
+    @bot.command()
+    async def search(ctx, *, term):
```

Comment:
all of this stuff was split into separate files for a reason. lumping them all back into `feubot.py` is just a huge maintenance burden